website/docs/feature-flag-tutorials/python/implementing-feature-flags.md

```diff
@@ -0,0 +1,331 @@
+
+
+## 6. Verify the toggle experience
+
+
+Now that we’ve added in new functionality and connected it to our feature flag, we can verify that if you disable the flag in your development environment, your Python app will no longer serve an HTML page with the option to delete surveys you’ve created.
+
+In your local Unleash instance, turn off the feature flag by disabling it in the development environment.
+
+![Image of feature flag with a disabled environment](/img/python-tutorial-disabled-flag.png)
+
+Next, return to your Survey app and refresh the browser. With the flag disabled, the delete button will no longer be visible.
+
+> **Note:** An update on the Python SDK may take around 30 seconds.
+
+![Screenshot of app in browser without delete buttons for surveys](/img/python-tutorial-surveys-without-delete.png)
+
+
+## 7. Improve a feature flag implementation with error handling
+
+If you turn the feature flag off, you won’t be able to use the delete button to remove a survey from your list. However, if a user wanted to bypass the UI to delete a survey, they could still use the URL from the delete method route to target a survey and delete it.
+
+They could do this because we have committed code that is only _partially_ hidden behind a feature flag. The HTML code is behind the flag, but the server method that it talks to is not.
+
+In a real world application, ignoring this would cause a user to perform an action they _shouldn’t_ able to. Luckily, we can use a feature flag to stop the delete method from being called manually.
+
+Let’s walk through how to gracefully handle this scenario:
+
+We need an error handler route to return a simple 404 page to stop a user from being able to manually delete a survey when the flag is off.
+
+In your `routes.py` file, import two more modules from Flask that will support our error handling function: [`abort`](https://flask.palletsprojects.com/en/3.0.x/api/#flask.abort) and [`jsonify`](https://flask.palletsprojects.com/en/3.0.x/api/#flask.json.jsonify).
+
+Line 1 will now look like this:
+
+```py
+from flask import redirect, render_template, request, url_for, abort, jsonify
+```
+
+Next, add in the following error handling method at the bottom of the file:
+
+```py
+@bp.errorhandler(404)
+def resource_not_found(e):
+   return jsonify(error=str(e)), 404
+```
+
+In order to render the error message, we can call it from the `delete_survey` method only in the case that the feature flag is turned off. Here’s how the updated `delete_survey` code would look like:
+
+```py
+@bp.route("/surveys/<int:survey_id>/delete", methods=["GET", "POST", "DELETE"])
+def delete_survey(survey_id):
+   if client.is_enabled('delete_survey_flag'):
+       survey = db.get_or_404(Survey, survey_id)
+       db.session.delete(survey)
+       db.session.commit()
+
+       return redirect(url_for("surveys.surveys_list_page"))
+   else:
+       abort(404, description="Resource not found")
```

Comment:
I would suggest the opposite if possible. Guard the route with a simple check: "If not enabled, abort". Otherwise, proceed with the normal behavior.